daemon/graphdriver/overlay2/check.go

```diff
@@ -100,3 +100,48 @@ func doesSupportNativeDiff(d string) error {
 
 	return nil
 }
+
+// supportsMultipleLowerDir checks if the system supports multiple lowerdirs,
+// which is required for the overlay2 driver. On 4.x kernels, multiple lowerdirs
+// are always available (so this check isn't needed), and backported to RHEL and
+// CentOS 3.x kernels (3.10.0-693.el7.x86_64 and up). This function is to detect
+// support on those kernels, without doing a kernel version compare.
+func supportsMultipleLowerDir(d string) error {
+	td, err := ioutil.TempDir(d, "multiple-lowerdir-check")
+	if err != nil {
+		return err
+	}
+	defer func() {
+		if err := os.RemoveAll(td); err != nil {
+			logrus.Warnf("Failed to remove check directory %v: %v", td, err)
+		}
+	}()
+
+	if err := os.MkdirAll(filepath.Join(td, "lower1"), 0755); err != nil {
+		return err
+	}
+	if err := os.MkdirAll(filepath.Join(td, "lower2"), 0755); err != nil {
+		return err
+	}
+	if err := os.Mkdir(filepath.Join(td, "upper"), 0755); err != nil {
+		return err
+	}
+	if err := os.Mkdir(filepath.Join(td, "work"), 0755); err != nil {
+		return err
+	}
+	if err := os.Mkdir(filepath.Join(td, "merged"), 0755); err != nil {
+		return err
+	}
+
+	opts := fmt.Sprintf("lowerdir=%s:%s,upperdir=%s,workdir=%s", path.Join(td, "lower2"), path.Join(td, "lower1"), path.Join(td, "upper"), path.Join(td, "work"))
+	if err := unix.Mount("overlay", filepath.Join(td, "merged"), "overlay", 0, opts); err != nil {
+		return errors.Wrap(err, "failed to mount overlay")
```

Comment:
What is the error that gets returned when it is not supported?

Reply:
I saw @kolyshkin tested this;

CentOS version | Kernel version | Compile date | Error
--- | --- | --- | ---
7.1 | 3.10.0-229.1.2.el7.x86_64 | ? | failed to mount overlay: invalid argument
7.1 | 3.10.0-229.20.1.el7.x86_64 | ? | failed to mount overlay: invalid argument
7.2 | 3.10.0-327.el7.x86_64 | Nov 19 2015 | nil
7.2 | 3.10.0-327.36.3.el7.x86_64 | Oct 24 2016 | nil
7.3 | 3.10.0-514.26.2.el7.x86_64 | Jul 4 2017  | nil